### PR TITLE
Minishift tests

### LINKED ISSUE #4288: Use PSI for running OpenShift 3.x tests
OpenShift 3.11 tests that were earlier working on Travis need to run on PSI infrastructure. This is mainly required to test Service Catalog.

/area testing
/priority high

Work pending for Sprint 198:
The required scrips are pending review in PR https://github.com/openshift/odo/pull/4406. Once merged the corrsponding Jenkins job will be enabled to complete the setup for first run.

**Acceptance criteria:**
- [x] Trigger Jenkins job to run tests in Minishift
- [x] E2E and integration tests are executed
- [ ] Tests are running correctly. This means that they are not flaking. If the test suite passes, then the repeated runs with the same code base pass as well.

### PR DESCRIPTION
**Pull request by rnapoles-rh**
**What type of PR is this?**
/kind failing-test

**What does does this PR do / why we need it**:
Removing unnecessary change of directory and merge start if minishift required into minikube-minishift-setup-env.sh

**Which issue(s) this PR fixes**:

Fixes #4288 

**PR acceptance criteria**:

- [x] Unit test 

- [ ] Integration test 

- [ ] Documentation 

- [x] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:
Check that ci/prow/psi-minishift-integration-e2e tests run from beginning to end.

### CODE REVIEW

**Comment by rnapoles-rh** [comment]
Not sure why ci/prow/psi-minishift-integration-e2e 
A job 
https://main-jenkins-csb-devtoolsodo.cloud.paas.psi.redhat.com/job/odo-minishift-pr-build/302/console

**Comment by rnapoles-rh** [comment]
This PR fix the following problems and improves error messages:
```
[ssh:Fedora 32] ++ sh .scripts/minishift-start-if-required.sh
[ssh:Fedora 32] sh: .scripts/minishift-start-if-required.sh: **No such file or directory**
[ssh:Fedora 32] + . ./scripts/minikube-minishift-all-tests.sh minishift
[ssh:Fedora 32] ++ set -ex
[ssh:Fedora 32] ++ case ${1} in
[ssh:Fedora 32] ++ cd /home/fedora/openshift/odo
[ssh:Fedora 32] ./scripts/minikube-minishift-all-tests.sh: line 19: cd: /home/fedora/openshift/odo: `No such file or directory`

```

File: scripts/minikube-minishift-setup-env.sh
Comment (mohammedzee1000) [inline scripts/minikube-minishift-setup-env.sh:9]: I would not want to expose this information on prow side. Jenkins should already be aware of the hostname

Comment (mohammedzee1000) [inline scripts/minikube-minishift-setup-env.sh:9]: I would not want to expose this information on prow side. Jenkins should already be aware of the hostname

File: scripts/minikube-minishift-setup-env.sh
Comment (mohammedzee1000) [inline scripts/minikube-minishift-setup-env.sh:64]: Would you not need to read `msstatus` again, if the previous status was `command not found`? If the minishift command does not exist this will never get to start the vm

File: scripts/minikube-minishift-setup-env.sh
Comment (mohammedzee1000) [inline scripts/minikube-minishift-setup-env.sh:83]: This logic looks good

**Review by mohammedzee1000 — CHANGES_REQUESTED** [review]

**Comment by prietyc123** [comment]
> some test failed but that is another story and will investigate why they are failing an will open another issue

If I am not wrong fixing tests also is part of #4288 , no?

**Comment by mohammedzee1000** [comment]
> > some test failed but that is another story and will investigate why they are failing an will open another issue
> 
> If I am not wrong fixing tests also is part of #4288 , no?

That has already been referenced in the description

File: scripts/minikube-minishift-setup-env.sh
Comment (rnapoles-rh) [reply scripts/minikube-minishift-setup-env.sh:9]: I will remove it, a debug message left behind

File: scripts/minikube-minishift-setup-env.sh
Comment (rnapoles-rh) [reply scripts/minikube-minishift-setup-env.sh:64]: will get msstatus again

**Comment by rnapoles-rh** [comment]
> > some test failed but that is another story and will investigate why they are failing an will open another issue
> 
> If I am not wrong fixing tests also is part of #4288 , no?

If we get this merged then we can get e2e and integration tests running for minishift and will be able to see in the logs which tests are failing. At this point we can not get the executed as per https://github.com/openshift/odo/pull/4561#issuecomment-811407267
I will open another issue for failing tests.

**Review by mohammedzee1000 — APPROVED** [review]
Looks good to me
/lgtm

**Comment by prietyc123** [comment]
> @prietyc123  No, #4288 is "Use PSI for running OpenShift 3.x tests", one thing is to run them, a different one is to have all tests passing. If we get this merged then we can get e2e and integration tests running for minishift and will be able to see in the logs which tests are failing.

I remember As per the discussion with @kadel #4288 will be said to be fixed iff all the tests go green on the pr with logs. Yes, this pr could be said as a part of #4288 being fixed. @kadel correct me if I am wrong.

**Comment by prietyc123** [comment]
> That has already been referenced in the description

But the description says that it will fix 4288, so once this pr gets merged it will close 4288. IIRC the scope of 4288 was green prow job for minishift, no?

**Comment by prietyc123** [comment]
Holding this pr until things get clear on whether this pr fix 4288 or not. @kadel Could you please share your thoughts.

/hold

**Comment by kadel** [comment]
> I remember As per the discussion with @kadel #4288 will be said to be fixed iff all the tests go green on the pr with logs.

Not sure if I'm not missing something, but the test are still failing. 
```
ERRO[2021-04-06T11:01:40Z] [ssh:Fedora 32] ./scripts/minikube-minishift-all-tests.sh: line 21: oc: command not found 
```

**Comment by prietyc123** [comment]
> Not sure if I'm not missing something, but the test are still failing.

@kadel So just wanted to confirm #4288 should be closed iff we have all the tests running on minishift, right?

**Comment by kadel** [comment]
> @kadel So just wanted to confirm #4288 should be closed iff we have all the tests running on minishift, right?

successfully running!

**Comment by kadel** [comment]
> @prietyc123 No, #4288 is "Use PSI for running OpenShift 3.x tests", one thing is to run them, a different one is to have all tests passing. If we get this merged then we can get e2e and integration tests running for minishift and will be able to see in the logs which tests are failing. At this point we can not get the executed as per [#4561 (comment)](https://github.com/openshift/odo/pull/4561#issuecomment-811407267)
> I will open another issue for failing tests.


Currently, tests are not even correctly executed :/ https://github.com/openshift/odo/pull/4561#issuecomment-814659748

If there are a few tests that are failing for some reasons that are outside of your control we can consider addressing them separately. 

But clicking randomly through psi-minishift-integration-e2e logs I didn't see any runs where it would get into the phase where it actually executes tests. Few instances that I saw it was always falling before tests were even executed.

**Comment by rnapoles-rh** [comment]
> > @prietyc123 No, #4288 is "Use PSI for running OpenShift 3.x tests", one thing is to run them, a different one is to have all tests passing. If we get this merged then we can get e2e and integration tests running for minishift and will be able to see in the logs which tests are failing. At this point we can not get the executed as per [#4561 (comment)](https://github.com/openshift/odo/pull/4561#issuecomment-811407267)
> > I will open another issue for failing tests.
> 
> Currently, tests are not even correctly executed :/ [#4561 (comment)](https://github.com/openshift/odo/pull/4561#issuecomment-814659748)
> 
> If there are a few tests that are failing for some reasons that are outside of your control we can consider addressing them separately.
> 
> But clicking randomly through psi-minishift-integration-e2e logs I didn't see any runs where it would get into the phase where it actually executes tests. Few instances that I saw it was always falling before tests were even executed.

@kadel since the ci-firewall was running test in non-intended target VMs the tests failed. When executed on the correct target VM where minishift is installed all tests were executed. Anyway, I will open issues for failing tests and that the failure is not related to the PSI setup and will list them in #4288

File: scripts/minici-presubmit-all-tests.sh
Comment (prietyc123) [inline scripts/minici-presubmit-all-tests.sh:57]: you need to add `--sendexchange` inspite of `--exchange` .

**Review by prietyc123 — CHANGES_REQUESTED** [review]

**Comment by prietyc123** [comment]
Minikube job fails due to `odo push` throwing

```
12:26:21.002673   88758 adapter.go:476] The component name is wyxvvb\n[ssh:Fedora 32] [odo]  ✗  Failed to start component with name wyxvvb. Error: Failed to create the component: unable to create or update component: unable to create Deployment wyxvvb: the namespace of the provided object does not match the namespace sent on the request\n[ssh:Fedora
```
which is just a flake. Other than that we have a successful run on minikube.

Logs: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_odo/4561/pull-ci-openshift-odo-master-psi-kubernetes-integration-e2e/1380123412982140928

/test psi-kubernetes-integration-e2e

**Comment by kadel** [comment]
> Minikube job fails due to `odo push` throwing
> 
> ```
> 12:26:21.002673   88758 adapter.go:476] The component name is wyxvvb\n[ssh:Fedora 32] [odo]  ✗  Failed to 
> start component with name wyxvvb. Error: Failed to create the component: 
> unable to create or update component: unable to create Deployment wyxvvb: 
> the namespace of the provided object does not match the namespace sent on the request\n[ssh:Fedora
> ```
> 
> which is just a flake. Other than that we have a successful run on minikube.
> 
> Logs: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_odo/4561/pull-ci-openshift-odo-master-psi-kubernetes-integration-e2e/1380123412982140928
> 
> /test psi-kubernetes-integration-e2e

The logging messages are horribly unreadable. We need to do something with it.

**Comment by prietyc123** [comment]
> The logging messages are horribly unreadable. We need to do something with it.

Yes, I will look into it.

**Comment by mohammedzee1000** [comment]
It is klog messages that need to be improved

**Comment by dharmit** [comment]
@rnapoles-rh is this PR still relevant considering we're working on #4800?

**Comment by dharmit** [comment]
Since OCP 3.11 is no longer supported, we don't need to put more efforts on this. Closing this PR.

/close

**Pull request closed without merging**
